models/sales/intermediate/int_hubspot__deals_enhanced.sql

```diff
@@ -44,11 +59,17 @@ with deals as (
         on deals.deal_pipeline_id = pipelines.deal_pipeline_id
     left join pipeline_stages 
         on deals.deal_pipeline_stage_id = pipeline_stages.deal_pipeline_stage_id
+    left join aggregate_merged_deals
+        on deals.deal_id = aggregate_merged_deals.deal_id
+    left join merged_deals
+        on deals.deal_id = merged_deals.merged_deal_id
 
     {% if var('hubspot_owner_enabled', true) %}
     left join owners 
         on deals.owner_id = owners.owner_id
     {% endif %}
+
+    where merged_deals.merged_deal_id is null -- remove deals that have been merged
```

Comment:
Is this true? I don't imagine the `merged_deals` table will contain all `deal_id`s and then if it has no merged deals then it will be `null` for the `merged_deal_id` column. I would assume this table is only populated with deals that have been merged. Therefore, if a deal has not been merged I would assume it would not be present in this table. 

If the above is the case then this logic would not be accurate for filtering out merged deals.

Reply:
When looking at the underlying data I see that there do in fact seem to be more deals in the `deal` table than exist in the `merged_deal` table. Given this, I believe this logic needs to be adjusted to accurately filter out merged deals.

![image](https://github.com/fivetran/dbt_hubspot/assets/74217849/1fb0a017-b2fa-4a69-ad33-161ef473667e)

Reply:
Also is filtering out the `null` records the correct way to filter out merged accounts? Wouldn't we want to cross reference which deals are present in the `merged_deal_id` column and then when there are any matches that is what we filter out?

Let me know if I am missing a component, but I am not sure if filtering where the merged_deal_id is null is the right way to achieve this.

Reply:
Oh goodness yes I confused how I was looking at the table-- right not all deals exist in the `merged_deals`. Therefore to correctly filter out deals in the final models that have been merged, we would need to do a check for if `deal_id` is a `merged_deal_id`. 

I updated it to the following:

`where deals.deal_id not in (select merged_deal_id from merged_deals) `